crates/core/tedge_watchdog/src/systemd_watchdog.rs

```diff
@@ -242,27 +251,55 @@ mod tests {
         let (mut sender, mut receiver) = mpsc::unbounded::<Message>();
         let health_topic = Topic::new("tedge/health/test-service").expect("Valid topic");
 
-        for x in 1..5i64 {
+        for _x in 1..5i64 {
             let health_status = json!({
                 "status": "up",
                 "pid": 123u32,
-                "time": x,
+                "time": "2013-06-22T17:03:14.000+02:00",
             })
             .to_string();
             let health_message = Message::new(&health_topic, health_status);
             sender.publish(health_message).await?;
         }
 
-        let health_status = get_latest_health_status_message(3, &mut receiver).await;
-        assert_eq!(health_status.time, 3);
+        let health_status =
+            get_latest_health_status_message("2013-06-21T17:03:14.000+02:00".into(), &mut receiver)
+                .await;
+
+        assert_eq!(
+            &health_status.unwrap().time,
+            "2013-06-22T17:03:14.000+02:00"
+        );
```

Comment:
That changed the logic of the test. The test was validating that if it passes the 3rd timestamp, then the response won't have the older timestamps: 1 & 2. This modified just tests the equality aspect.